src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -24,6 +24,7 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 			}
 			BaseType = baseType;
 			Interfaces = interfaces ?? new System.Type[0];
+			_uniqueInterfaces = new HashSet<System.Type>(Interfaces);
```

Comment:
> Ok, no it can not, as hashCode is calculated using just xor (which is strange as there will be lots of collisions)

When used to combine hashcode from a set (so different objects), it should not be that bad. It is not like `&` or `|` which ends up zeroing or "one-ing" all bits. It seems it is even a [common practice in Java](http://stackoverflow.com/a/19139500/1178314).

Now if we want to change it for a more classical approach like multiplying by a prime then adding next value while overflowing, we will need to take care of the order.